src/ProjectTemplates/Web.ProjectTemplates/content/Components-CSharp/.template.config/template.json

```diff
@@ -0,0 +1,287 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "author": "Microsoft",
+  "classifications": [
+    "Web",
+    "Blazor",
+    "WebAssembly",
+    "PWA"
+  ],
+  "name": "Blazor Web App",
+  "defaultName": "BlazorApp",
+  "description": "A project template for creating a Blazor app hosted inside an ASP.NET app that runs on the server. This template can be used for web apps with rich dynamic user interfaces (UIs).",
+  "groupIdentity": "Microsoft.Web.Blazor",
+  "precedence": "9800",
+  "guids": [
+    "4C26868E-5E7C-458D-82E3-040509D0C71F",
+    "5990939C-7E7B-4CFA-86FF-44CA5756498A",
+    "650B3CE7-2E93-4CC4-9F46-466686815EAA",
+    "0AFFA7FD-4E37-4636-AB91-3753E746DB98"
+  ],
+  "identity": "Microsoft.Web.Blazor.CSharp.8.0",
+  "thirdPartyNotices": "https://aka.ms/aspnetcore/8.0-third-party-notices",
+  "preferNameDirectory": true,
+  "primaryOutputs": [
+    {
+      "path": "Components-CSharp.csproj"
+    }
+  ],
+  "shortName": "blazor",
+  "sourceName": "Components-CSharp",
+  "sources": [
+    {
+      "source": "./",
+      "target": "./",
+      "exclude": [
+        ".template.config/**"
+      ],
+      "copyOnly": [
+        "**/wwwroot/css/**"
+      ],
+      "modifiers": [
+        {
+          "condition": "(!PWA)",
+          "exclude": [
+            "wwwroot/service-worker*.js",
+            "wwwroot/manifest.webmanifest",
+            "wwwroot/icon-512.png"
+          ]
+        },
+        {
+          "condition": "(!UseClient)",
+          "exclude": [
+            "Program.Browser.Main.cs",
+            "Program.Browser.cs",
+            "wwwroot/appsettings.Development.json",
+            "wwwroot/appsettings.json"
+          ],
+          "rename": {
+            "Program.Server.Main.cs": "Program.cs",
+            "Program.Server.cs": "Program.cs"
+          }
+        },
+        {
+          "condition": "(!UseProgramMain)",
+          "exclude": [
+            "Program.Server.Main.cs",
+            "Program.Browser.Main.cs"
+          ]
+        },
+        {
+          "condition": "(UseProgramMain && !UseClient)",
+          "exclude": [
+            "Program.Server.cs",
+            "Program.Browser.cs"
+          ],
+          "rename": {
+            "Program.Server.Main.cs": "Program.cs"
+          }
+        },
+        {
+          "condition": "(UseProgramMain && UseClient)",
+          "exclude": [
+            "Program.Server.cs",
+            "Program.Browser.cs"
+          ],
+          "rename": {
+            "Program.Server.Main.cs": "Program.Server.cs",
+            "Program.Browser.Main.cs": "Program.Browser.cs"
```

Comment:
Another terminology decision to make. I know the TFM is `-browser`, but we are focused on the term WebAssembly in most other places (e.g., product naming, render modes). Arguably the TFM is really just shorthand for `-wasm-browser`, as browser is understood as a flavour of WebAssembly.

I personally am more drawn to a `*.WebAssembly.cs` naming convention here. What do you think?

Reply:
From a completionist perspective, one could imagine some future day when there's a TFM for `-wasi` or another WebAssembly flavour. In that case, you might expect:

 * `Program.cs` - included in all compilations
 * `Program.WebAssembly.cs` - included in all WebAssembly-type compilations (browser and wasi)
 * `Program.Browser.cs` - included only in `-browser`

If that model were used, it would be legit to have either `Program.WebAssembly.cs` or `Program.Browser.cs` in this template, so we are free to choose either option right now. I'm still a bit more drawn to .WebAssembly.

Reply:
I'd rather to `.Browser` since that aligns with the tfm and matches the expectations across dotnet. Choosing a special value means all the tooling stack/libraries need to special case the browser tfm instead of treating it like other TFMs.

Reply:
I am settling this for now as we will maintain .browser and .server. But let's have a separate conversation offline.